### PR TITLE
Merge v1.0.0 changes to develop

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,9 @@ cmake_policy(SET CMP0057 NEW)
 
 project(Umpire
   LANGUAGES CXX C
-  VERSION 0.3.5)
+  VERSION 1.0.0)
+
+set(UMPIRE_VERSION_RC "rc1")
```
Comment (davidbeckingsale) [inline CMakeLists.txt]: In `develop` I will make another PR to set this to be the git hash.

File: scripts/make_release_tarball.sh [hunk 2]
```diff
@@ -7,7 +7,7 @@
 ##############################################################################
 
 TAR_CMD=gtar
-VERSION=0.3.5
+VERSION=1.0.0-rc1
```
Comment (davidbeckingsale) [inline scripts/make_release_tarball.sh]: This needs to be changed to remove the rc1 prior to release.

File: CMakeLists.txt (hunk 1 above)
Comment (mcfadden8) [reply CMakeLists.txt]: @davidbeckingsale, when will this other PR be created?  After this is merged into develop?

Comment (davidbeckingsale) [reply CMakeLists.txt]: UM-426

**Pull request merged**